File: js/hang/src/catalog/preview.ts
import { z } from "zod";

export const PreviewSchema = z.object({
	name: z.string().optional(), // name
	avatar: z.string().optional(), // avatar

	audio: z.boolean().optional(), // audio enabled
	video: z.boolean().optional(), // video enabled

	speaking: z.boolean().optional(), // actively speaking
	typing: z.boolean().optional(), // actively typing
	chat: z.boolean().optional(), // chatted recently
	screen: z.boolean().optional(), // screen sharing
});

export type Preview = z.infer<typeof PreviewSchema>;

File: rs/hang/src/catalog/preview.rs
use serde::{Deserialize, Serialize};

/// Preview information about a broadcast
#[serde_with::skip_serializing_none]
#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Default)]
#[serde(rename_all = "camelCase")]
pub struct Preview {
	pub name: Option<String>,   // name
	pub avatar: Option<String>, // avatar

	pub audio: Option<bool>,  // audio enabled
	pub video: Option<bool>,  // video enabled
	pub screen: Option<bool>, // screen sharing

	pub speaking: Option<bool>, // actively speaking
	pub typing: Option<bool>,   // actively typing
	pub chat: Option<bool>,     // chatted recently
}
